### PR TITLE
Add module for doing iron-birthweight back-of-envelope calculations

### PR DESCRIPTION
**Pull request by NathanielBlairStahn**

### CODE REVIEW

File: nanosim_models/lbwsg_boe.py [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+#     for propensity_name in propensity_names:
+#         assign_simulant_property(pop, propensity_name)
+
+class IronBirthweightCalculator:
+    """Class to run nanosimulations for the effect of iron on low birthweight."""
+    
+    treated_lbwsg_rr_colname = 'treated_lbwsg_rr'
+
+    def __init__(self, location, artifact_path, year, draws, vehicle, covered_proportion_of_eats_fortifiable,
+                 take_mean=False, risk_effect_class=lbwsg.LBWSGRiskEffect, random_seed=None):
```
Comment (KjellSwedin) [inline nanosim_models/lbwsg_boe.py:69]: Shouldn't the seed default to something known?

File: nanosim_models/lbwsg_boe.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+        # Save input parameters so we can look them up later if necessary/desired
+        self.location = location
+        self.artifact_path = artifact_path
+        self.year = year
+#         self.draws = draws # These will also be stored in global_data, unless take_mean is True
+        
+        # TODO: Perhaps create and save a numpy random generator, and share it via global_data
+        # random_generator = np.random.default_rng(random_seed) # Now do something with this...
+        np.random.seed(random_seed)
+
```
Comment (KjellSwedin) [inline nanosim_models/lbwsg_boe.py:81]: Your TODO is what the numpy docs recommend. They recommend against re-seeding the default generator

**Review by KjellSwedin — APPROVED** [review]
Context regarding the state of the code would be helpful for reviewers. This can just be a comment on the PR. For instance, I don't really know what to do with all of the commented out code. If I _might_ use code in the future I will sometimes check it in, to have it in source control, then follow up with a checkin that deletes it.

Be deliberate about functions with side effects. You generally do a nice job naming functions and variables which really helps readability. However, even if one can infer what is going on by the name, non-pure functions (functions with side effects) are harder to reason about and usually harder to debug (https://en.wikipedia.org/wiki/Side_effect_(computer_science)).

File: nanosim_models/lbwsg_boe.py (hunk 1 above)
Comment (NathanielBlairStahn) [reply nanosim_models/lbwsg_boe.py:69]: Hmm, I've been setting the seed farther out in the code hierarchy, e.g. in functions that use this class, but maybe it makes more sense to set a default here?  I'll consider changing it.

File: nanosim_models/lbwsg_boe.py (hunk 2 above)
Comment (NathanielBlairStahn) [reply nanosim_models/lbwsg_boe.py:81]: Yep, that's why I added the TODO, but I'll need to go through my code and update all the functions that use randomness to accept a Generator instance as an argument, and I haven't had a chance to do that yet.

**Comment by NathanielBlairStahn** [comment]
> Context regarding the state of the code would be helpful for reviewers. This can just be a comment on the PR. For instance, I don't really know what to do with all of the commented out code. If I _might_ use code in the future I will sometimes check it in, to have it in source control, then follow up with a checkin that deletes it.

Sorry my PRs are such a mess. This entire barrage of PRs is just copying existing code from the `vivarium_data_analysis` repo, so maybe I didn't actually need to request reviewers, but I appreciate your comments, @KjellSwedin .

I haven't quite gotten the hang of a good git workflow that lets me get working code for a deadline, keep extraneous things out of the main branch, and create easily digestible PRs without spending an inordinate amount of time organizing everything and cleaning it up after I get something that works.  I'd be interested in learning from people with more experience about how to do this better.

**Comment by NathanielBlairStahn** [comment]
> Be deliberate about functions with side effects. You generally do a nice job naming functions and variables which really helps readability. However, even if one can infer what is going on by the name, non-pure functions (functions with side effects) are harder to reason about and usually harder to debug (https://en.wikipedia.org/wiki/Side_effect_(computer_science)).

As you noted in my other PR, maybe it's worth rethinking the entire design to see whether the functions that modify the population table in place are worth it.  For now I'll just try to make it clear what's going on until I have time to revise everything more thoroughly.

**Pull request merged**